java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java

```diff
@@ -424,9 +434,28 @@ private void parseImpl(
                     break;
                 default:
                     init (snapshot, task, false);
+                    DiagnosticListener<JavaFileObject> diagnosticListener;
+                    JavacTaskImpl javacTask;
+                    boolean oneInstanceJava=Boolean.getBoolean("java.enable.single.javac") && this.snapshotSize <= this.MAX_FILE_SIZE;
+                    if (sequentialParsing == null && ciImpl == null && oneInstanceJava) {
+                        List<JavaFileObject> jfos = new ArrayList<>();
+                        for (Snapshot s : snapshots) {
+                            jfos.add(FileObjects.sourceFileObject(s.getSource().getFileObject(), root, JavaFileFilterQuery.getFilter(s.getSource().getFileObject()), s.getText()));
+                        }
+                        diagnosticListener = new CompilationInfoImpl.DiagnosticListenerImpl(this.root, jfos.get(0), this.cpInfo);
+                        javacTask = JavacParser.createJavacTask(this.file, jfos, this.root, this.cpInfo,
+                                this, diagnosticListener, false);
+                    } else if (ciImpl != null &&  oneInstanceJava) {
+                        diagnosticListener = ciImpl.getDiagnosticListener();
+                        javacTask = ciImpl.getJavacTask();
+                        oldParsedTrees=ciImpl.getParsedTrees();
+                    } else {
+                        diagnosticListener = null;
+                        javacTask = null;
+                    }
                     ciImpl = createCurrentInfo(this, file, root, snapshot,
-                        sequentialParsing == null || ciImpl == null ? null : ciImpl.getJavacTask(),
-                        sequentialParsing == null || ciImpl == null ? null : ciImpl.getDiagnosticListener());
+                        javacTask,
```

Comment:
I suspect this will disable "sequential" processing when nb-javac is installed, right? Or do I miss something here?

Reply:
This is handled above on the if-else conditions starting at line 440. If nb-javac is installed then sequential parsing will not be null and remaining code will be same as of before this PR.

Reply:
Is this OK @jlahoda, can this PR be merged now??

Reply:
I note that the if on line 448 is also checking oneInstanceJava. So it seems to me that is sequentialParsing != null and oneInstanceJava == false, the it will always fall to the else section and go file-by-file. Or what do I miss?

Reply:
You're correct @jlahoda removed the oninstance variable form second if.
Now this PR is final